### DISCUSSION TITLE
Realtime clock and realtime ETA

### DISCUSSION THREAD

**Discussion by kallestrop** [JAGFx/ets2-dashboard-skin#101] [Ideas]
Hello,

I have a suggestion: 

Is it  possible to add a real time clock and a estimated time of arrival (ETA) in real time to the dashboard?

It could be located on the top line next to the ingame time with an option to enable/disable.

Until now this information is only available in 3rd party apps like Trucky overlay or SpedV.

**Comment by JAGFx** [comment]
Hello,

Thanks for your interest.

Adding a real time clock is possible, but a real time related to the actual work will be complicated.

The game time system is different from real time. So using both to get the ETA will create an irrelevant time.
We not have all data like plugin used directly in game. The SCS software telemetry (plugin used for external app like this project) do not provide all necessary data
Eventually, we can use both when the current job is in WOT.

But again, we have a problem with the time zone. See [this issue](https://github.com/JAGFx/ets2-dashboard-skin/issues/33#issuecomment-725514001). ;)

In summary, adding real time will be a great feature, but adding ETA based on real time will not be possible and relevant yet.

**Comment by kallestrop** [comment]
Hello,

thanks for the explanation. Regarding the real time eta, i thought about a simple rule of three calculation with the remaining distance. So you don´t need time information from ingame.

For example: Polling remaining distance every 60 seconds:

read remaining distance1 --> wait 60 seconds --> read remaining distance2 --> calculate distance1 - distance2 = covered track in 60 seconds 

So the calculation with a covered track of 25 Km in 60 seconds and a remaining distance of 2500 Km and a current time of 20:00 would be:

2500 / 25 = 100

20:00 + 100 minutes = 21:40 eta in realtime

**Reply by JAGFx** [reply]
Hum, why not.

I let you create an issue and link this discussion.

I'll work on later.

Thanks ;)
